src/Resource.js

```diff
@@ -9,8 +9,15 @@ const getOptionsQuery = (optionsObject = {}) =>
     .map(k => `${k}=${encodeURIComponent(optionsObject[k])}`)
     .join('&');
 
-const relatedResourceUrl = ({ parent, relationship }) =>
-  `${parent.type}/${parent.id}/${relationship}`;
+const relatedResourceUrl = ({ parent, relationship }) => {
+  if (
+    parent.relationships &&
+    Object.keys(parent.relationships).includes(relationship)
+  ) {
+    return parent.relationships[relationship].links.related;
```

Comment:
From the [JSON:API spec](https://jsonapi.org/format/#document-resource-object-relationships) it looks like both the `links` and `related` keys are optional. Could you add a check for each of those two keys? That way if they are missing the code will still fall back on the hard-coded relationship URL.

Reply:
I just added some safe navigation code. Is it safe enough?

Reply:
Looks good!